pkg/eks/components/CruEKS.vue

```diff
@@ -0,0 +1,766 @@
+      type:    String,
+      default: _CREATE
+    },
+
+    // v2 provisioning cluster object
+    value: {
+      type:    Object,
+      default: () => {
+        return {};
+      }
+    }
+  },
+
+  async fetch() {
+    const store = this.$store as Store<any>;
+
+    if (this.value.id) {
+      const liveNormanCluster = await this.value.findNormanCluster();
+
+      this.normanCluster = await store.dispatch(`rancher/clone`, { resource: liveNormanCluster });
+      // track original version on edit to ensure we don't offer k8s downgrades
+      this.originalVersion = this.normanCluster?.eksConfig?.kubernetesVersion;
+    } else {
+      this.normanCluster = await store.dispatch('rancher/create', { type: NORMAN.CLUSTER, ...defaultCluster }, { root: true });
+    }
+
+    if (!this.normanCluster.eksConfig) {
+      this.$set(this.normanCluster, 'eksConfig', { ...DEFAULT_EKS_CONFIG });
+    }
+    if (!this.normanCluster.fleetAgentDeploymentCustomization) {
+      this.$set(this.normanCluster, 'fleetAgentDeploymentCustomization', {});
+    }
+    if (!this.normanCluster.clusterAgentDeploymentCustomization) {
+      this.$set(this.normanCluster, 'clusterAgentDeploymentCustomization', {});
+    }
+    this.config = this.normanCluster.eksConfig;
+
+    if (!this.config.nodeGroups || !this.config.nodeGroups.length) {
+      this.$set(this.config, 'nodeGroups', [{ ...DEFAULT_NODE_GROUP_CONFIG, nodegroupName: 'group1' }]);
+    }
+    this.nodeGroups = this.config.nodeGroups;
+    this.fetchInstanceTypes();
+    this.fetchLaunchTemplates();
+    this.fetchServiceRoles();
+  },
+
+  data() {
+    return {
+      cloudCredentialId: '',
+      normanCluster:     { name: '' } as any,
+      nodeGroups:        [] as EKSNodeGroup[],
+      config:            { } as EKSConfig,
+      membershipUpdate:  {} as any,
+      originalVersion:   '',
+
+      fvFormRuleSets: [{
+        path:  'name',
+        rules: ['nameRequired'],
+      },
+      {
+        path:  'nodegroupNames',
+        rules: ['nodeGroupNamesUnique', 'nodeGroupNamesRequired']
+      },
+      {
+        path:  'maxSize',
+        rules: ['maxSize']
+      },
+      {
+        path:  'minSize',
+        rules: ['minSize']
+      },
+      {
+        path:  'desiredSize',
+        rules: ['desiredSize']
+      },
+      {
+        path:  'subnets',
+        rules: ['subnets']
+      },
+      {
+        path:  'instanceType',
+        rules: ['instanceType']
+      },
+      {
+        path:  'diskSize',
+        rules: ['diskSize']
+      },
+      {
+        path:  'networking',
+        rules: ['publicPrivateAccess']
+      },
+      ],
+
+      loadingInstanceTypes:   false,
+      loadingLaunchTemplates: false,
+
+      loadingIam:      false,
+      iamInfo:         {} as any,
+      ec2Roles:        [],
+      eksRoles:        [],
+      instanceTypes:   [],
+      launchTemplates: []
+    };
+  },
+
+  created() {
+    const registerAfterHook = this.registerAfterHook as Function;
```

Comment:
const registerAfterHook: Function = this.registerAfterHook;

The `as` will disregards the type check of `this.registerAfterHook`. the function is in a mixin, but ts should be ok finding it.